hw3/src/main.ts

```diff
@@ -0,0 +1,86 @@
+import { Either, fromPromise, ap, right, left, getOrElse, flatten } from "./fp/either";
+import { pipe } from "./fp/utils";
+import { fetchClient, fetchExecutor } from "./fetching";
+import { ClientUser, ExecutorUser, Demand } from "./types";
+import { Maybe, fromNullable, isNone } from "./fp/maybe";
+
+type Response<R> = Promise<Either<string, R>>;
+
+const getExecutor = (): Response<ExecutorUser> => fromPromise(fetchExecutor());
+const getClients = (): Response<Array<ClientUser & { demands: Maybe<Array<Demand>> }>> =>
+  fromPromise(fetchClient()).then((response) =>
+    response.map((clients) =>
```

Comment:
Property 'map' does not exist on type 'Either<unknown, { name: string; demands: Demand[]; position: { x: number; y: number; }; reward: number; }[]>'.
  Property 'map' does not exist on type 'Left<unknown>'.

Reply:
I want to place map high order function

Reply:
Looks like response is already clients in your case.
```
const getClients = (): Response<Array<ClientUser>> =>
  fromPromise(fetchClient().then((clients) => clients.map((client) => {
    return {
      ...client,
     demand: fromNullable(client.demands)
    };
  })));
```